lib/apparmortest.pm

```diff
@@ -667,6 +670,81 @@ sub yast2_apparmor_cleanup {
     upload_logs("$audit_log");
 }
 
+# Create a test profile with name contains '('
+sub create_a_test_profile_name_is_special {
+    my ($testfile, $str) = @_;
+
+    my $testfile2 = "$testfile" . "$str";
+    assert_script_run("cp $testfile $testfile2");
+    assert_script_run("aa-autodep $testfile2");
+    assert_script_run("ll /etc/apparmor.d/ | grep $str");
+}
+
+# Refer to "https://bugs.launchpad.net/apparmor/+bug/1848227"
+# to create a test profile with content "local include above the '}'",
+# then run "aa-complain, aa-disable, aa-enforce, aa-logprof"
+# to verify all the commands should be succeeded
+sub test_profile_content_is_special {
+    my ($self, $cmd, $msg) = @_;
+    my $test          = "test_profile";
+    my $test_profile  = "/etc/apparmor.d/usr.sbin." . "$test";
+    my $local_profile = "/etc/apparmor.d/local/usr.sbin.cupsd";
+
+    # Create an empty local profile under "/etc/apparmor.d/local/"
+    assert_script_run("rm -rf $local_profile");
+    assert_script_run("touch $local_profile");
+
+    # Create a test profile under "/etc/apparmor.d/"
+    assert_script_run("echo '/usr/sbin/cupsd {' > $test_profile");
+    assert_script_run("echo '}' >> $test_profile");
+    assert_script_run("echo '#include <local/usr.sbin.cupsd>' >> $test_profile");
+
+    # Run aa-* commands and check the output
+    my $current_ver = script_output("rpm -q --qf '%{version}' apparmor-utils");
+
+    my $cmd1 = $cmd eq "aa-logprof" ? $cmd : "$cmd $test_profile";
+    my $ret  = script_run($cmd1, sub { m/$msg/ });
+    if ($ret == 0) {
+        if ("$cmd" eq "aa-disable") {
+            # The profile will not be listed out if disabled
+            my $ret = script_run("aa-status | grep $test");
+            if ($ret == 0) {
+                $self->result("fail");
+            }
+        }
+    }
+    elsif ($ret && ($current_ver lt "2.13.4")) {
+        record_soft_failure("jsc#SLE-15464: Update Apparmor to 2.13.4");
```

Comment:
The Jira feature https://jira.suse.com/browse/SLE-15464 is specific to 15-SP3 and should not be used as a soft-failure reference for older products. Either there is a bug report for older products – which you can create of course – or the behaviour needs to be accepted as is without raising a soft-fail

Reply:
> The Jira feature https://jira.suse.com/browse/SLE-15464 is specific to 15-SP3 and should not be used as a soft-failure reference for older products. Either there is a bug report for older products – which you can create of course – or the behaviour needs to be accepted as is without raising a soft-fail
Sure, you explanation is very reasonable. I will file a bsc# bug to track the soft failure.